src/test/java/org/broadinstitute/hellbender/tools/dataflow/pipelines/MarkDuplicatesDataflowIntegrationTest.java

```diff
@@ -51,5 +92,27 @@ public void testMarkDuplicatesDataflowIntegrationTestLocal(final File input, fin
 
         Assert.assertEquals(totalReads, totalExpected, "Wrong number of reads in output BAM");
         Assert.assertEquals(duplicateReads, dupsExpected, "Wrong number of duplicate reads in output BAM");
+
+        final MetricsFile<DuplicationMetrics, Comparable<?>> metricsOutput = new MetricsFile<>();
+        try {
+            metricsOutput.read(new FileReader(metricsFile));
+        } catch (final FileNotFoundException ex) {
+            System.err.println("Metrics file not found: " + ex);
+        }
+        Assert.assertEquals(metricsOutput.getMetrics().size(), metricsExpected.size(),
+                            "Wrong number of metrics with non-zero fields.");
+        for (int i = 0; i < metricsOutput.getMetrics().size(); i++ ){
```

Comment:
Is this ordering guaranteed or is it based on whatever arbitrary order the `PCollection` of `KV<Library,DuplicationMetric>` happened to be in?